src/Moq/StubbedPropertySetup.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD, and Contributors.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Moq
+{
+	internal sealed class StubbedPropertySetup : Setup
+	{
+		private object value;
+
+		public StubbedPropertySetup(Mock mock, LambdaExpression expression, MethodInfo getter, MethodInfo setter, object initialValue)
+			: base(originalExpression: null, mock, new PropertyAccessorExpectation(expression, getter, setter))
+		{
+			Debug.Assert(mock != null);
+			Debug.Assert(expression != null);
+			Debug.Assert(expression.IsProperty());
+			Debug.Assert(getter != null || setter != null);
```

Comment:
Those assertions are redundant, since `PropertyAccessorExpectation` and the base class' constructor already assert the same things.